packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -497,111 +511,107 @@ final _response = await ${isRootClient ? 'this' : '_rootClient'}.doRequest(
-            if (operation.responses!.length > 1) {
-              throw Exception('Can not work with multiple status codes right now');
+          for (final responseEntry in responses.entries) {
+            final response = responseEntry.key;
+            final statusCodes = responseEntry.value;
+            code.write(
+              'if (${statusCodes.map((final statusCode) => '_response.statusCode == $statusCode').join(' || ')}) {',
+            );
+
+            String? headersType;
+            String? headersValue;
+            if (response.headers != null) {
+              final identifier =
+                  '${tag != null ? toDartName(tag, uppercaseFirstCharacter: true) : null}${toDartName(operationId, uppercaseFirstCharacter: true)}Headers';
+              final result = resolveObject(
+                spec,
+                state,
+                identifier,
+                Schema(
+                  properties: response.headers!.map(
+                    (final headerName, final value) => MapEntry(
+                      headerName.toLowerCase(),
+                      value.schema!,
+                    ),
+                  ),
+                ),
+                isHeader: true,
+              );
+              headersType = result.name;
+              headersValue = result.deserialize('_response.responseHeaders');
             }
-            for (final responseEntry in operation.responses!.entries) {
-              final statusCode = responseEntry.key;
-              final response = responseEntry.value;
-              code.write('if (_response.statusCode == $statusCode) {');
-
-              String? headersType;
-              String? headersValue;
-              if (response.headers != null) {
-                final identifier =
-                    '${tag != null ? toDartName(tag, uppercaseFirstCharacter: true) : null}${toDartName(operationId, uppercaseFirstCharacter: true)}Headers';
-                final result = resolveObject(
+
+            String? dataType;
+            String? dataValue;
+            bool? dataNeedsAwait;
+            if (response.content != null) {
+              if (response.content!.length > 1) {
+                throw Exception('Can not work with multiple mime types right now');
+              }
+              for (final content in response.content!.entries) {
+                final mimeType = content.key;
+                final mediaType = content.value;
+
+                final result = resolveType(
                   spec,
                   state,
-                  identifier,
-                  Schema(
-                    properties: response.headers!.map(
-                      (final headerName, final value) => MapEntry(
-                        headerName.toLowerCase(),
-                        value.schema!,
-                      ),
-                    ),
+                  toDartName(
+                    '$operationId-response-${statusCodes.join('-')}-$mimeType',
```

Comment:
Does it even make sense to include the status code in the name or do we need it to avoid collisions?
I think `200201400401404` is quite bad to read.

Reply:
Need to do that in order to avoid collision. The other option would be using the index of the response, that would at least be more readable.